feat: Set recipe default image

CHANMYUNG opened the pull request:
- Add recipe default image 'recipe-default.png'
- Set imgUrl to "/img/recipe-default.png" when parameter imgUrl is null
  in the Recipe's constructor

Resolves: #98

CHANMYUNG committed:
feat: Set recipe default image

- Add recipe default image 'recipe-default.png'
- Set imgUrl to "/img/recipe-default.png" when parameter imgUrl is null
  in the Recipe's constructor

Resolves: #98

CHANMYUNG commented:
지금 recipe default image path가 하드코딩 되어있는데, 어디에 있어야할지 고민하다가 그냥 푸시했어요.

- Recipe 클래스 내에 상수로 두고 @JsonIgnore하기

밖에 생각이 안나는데.. 다른 방법 생각나시면 제안해주시면 감사하겠습니다.

Lahavas commented on src/main/java/com/woowahan/techcamp/recipehub/recipe/domain/Recipe.java:
```diff
@@ -46,7 +46,7 @@ public Recipe(Category category, User owner, String name, boolean completed, Lis
         this.recipeSteps = recipeSteps;
-        this.imgUrl = imgUrl;
+        this.imgUrl = (imgUrl == null) ? "/img/recipe-default.png" : imgUrl;
     }
```
여기서만 Recipe 초기화가 이루어질까요??

ryuneeee commented on src/main/java/com/woowahan/techcamp/recipehub/recipe/domain/Recipe.java:
```diff
@@ -46,7 +46,7 @@ public Recipe(Category category, User owner, String name, boolean completed, Lis
         this.completed = completed;
         this.recipeSteps = recipeSteps;
-        this.imgUrl = imgUrl;
+        this.imgUrl = (imgUrl == null) ? "/img/recipe-default.png" : imgUrl;
```
어... 이건 하드코딩하면 좀... 차라리 properties로 빼주세요.

Lahavas commented:
클래스 내에 상수가 Json으로 변환이 되나요??

CHANMYUNG committed:
feat: Make Recipe class to contains default image url

- Make the recipe class to contains default image url
- Use Recipe.DEFAULT_IMAGE_URL in the test cases
- Add test case for recipe's constructor call by image url is null

Resolves: #98

CHANMYUNG commented:
@Lahavas 피드백 반영했습니다. 좋네요 👍

CHANMYUNG committed:
Merge branch 'develop' into feature/recipe-default-image

Lahavas reviewed (approved):
승인합니다~

Lahavas committed:
Merge branch 'develop' into feature/recipe-default-image

ryuneeee reviewed (approved):
굿굿

ffaass commented on src/main/java/com/woowahan/techcamp/recipehub/recipe/domain/Recipe.java:
```diff
@@ -18,6 +18,8 @@
 @NoArgsConstructor
 public class Recipe extends AbstractEntity {
 
+    public static final String DEFAULT_RECIPE_IMAGE_URL = "/img/recipe-default.png";
```
Production 환경에서는 s3에 있는 이미지를 참조해야 하지 않나요?
properties로 빼거나 Config에서 반환하도록 하는 게 좋을 것 같아요

CHANMYUNG replied:
어... 이건 s3에 안올리는데용. hero-create 이미지같은 개념으로 보셔야할 것 같은데요?

ryuneeee replied:
S3에 올릴수도 있고, 아닐수도 있고 개인차이죵 `~_~`

ryuneeee replied:
인스턴스 접근 권한을 가진사람이 S3 접근 권한이 없을 수도 있으므로 배포 파일에 가지고 있는게 맞는거 같아욤

CHANMYUNG replied:
아 당연히 배포파일에 들고있어야 할거라고 생각했는데 그렇게 생각하실 수도 있겠군뇨

ffaass replied:
아하 그렇네요 ㅎㅎ 그럼 Approve하겠습니다!

CHANMYUNG committed:
Merge branch 'develop' into feature/recipe-default-image

ffaass committed:
Merge branch 'develop' into feature/recipe-default-image
